global_body_planner/src/planning_utils.cpp

```diff
@@ -646,43 +625,44 @@ bool getRandomLeapAction(const State &s, const Eigen::Vector3d &surf_norm,
 
 bool refineAction(const State &s, Action &a,
                   const PlannerConfig &planner_config) {
-  double z_f_leap = planner_config.H_NOM + 0.05;
-  double z_f_land = planner_config.H_NOM;
-
-  if (!refineStance(s, z_f_leap, LEAP_STANCE, a, planner_config)) return false;
+  if (!refineStance(s, LEAP_STANCE, a, planner_config)) return false;
 
   State s_leap = applyStance(s, a, LEAP_STANCE, planner_config);
 
-  if (!refineFlight(s_leap, z_f_leap, a.t_f, planner_config)) {
+  if (!refineFlight(s_leap, a.t_f, planner_config)) {
     return false;
   }
 
   State s_land = applyFlight(s_leap, a.t_f, planner_config);
 
-  if (!refineStance(s_land, z_f_land, LAND_STANCE, a, planner_config))
-    return false;
+  if (!refineStance(s_land, LAND_STANCE, a, planner_config)) return false;
 
   State s_final = applyStance(s_land, a, LAND_STANCE, planner_config);
   a.dz_f = s_final.vel[2];
 
   return true;
 }
 
-bool refineStance(const State &s, double z_f, int phase, Action &a,
+bool refineStance(const State &s, int phase, Action &a,
                   const PlannerConfig &planner_config) {
 #ifdef DEBUG_REFINE_STATE
   std::cout << "Starting stance refine, phase = " << phase << std::endl;
 #endif
 
-  // Define parameters
+  // Load parameters
   double &t_s = (phase == LEAP_STANCE) ? a.t_s_leap : a.t_s_land;
   GRF &grf_stance = (phase == LEAP_STANCE) ? a.grf_0 : a.grf_f;
   double dz_0 = (phase == LEAP_STANCE) ? a.dz_0 : s.vel[2];
   double g = planner_config.G_CONST;
-  bool grf_valid, friction_cone_valid, final_state_valid, midstance_state_valid;
+#ifdef DEBUG_REFINE_STATE
+  std::cout << "dz_0 = " << dz_0 << std::endl;
```

Comment:
Should this be ROS_INFO?

Reply:
Those are just compiler macros I've defined to easily debug certain sections of the code. There are too many levels I need depending on what I'm debugging to rely on ROS debug levels, and I've tried to stay ROS agnostic.